js/preview/src/examine.py

```diff
@@ -9,58 +9,67 @@
 from gradio.blocks import BlockContext
 from gradio.components import Component
 
-parser = argparse.ArgumentParser(description="Description of your program")
-parser.add_argument("-m", "--mode", help="Build mode or dev mode")
-args = parser.parse_args()
-
-with open("../pyproject.toml") as f:
-    pyproject_source = f.read()
-
-pyproject_toml = parse(pyproject_source)
-module_name = pyproject_toml["project"]["name"]
-module = importlib.import_module(module_name)
-
-artifacts: list[str] = pyproject_toml["tool"]["hatch"]["build"]["artifacts"]
-
-
-def get_relative_path(path):
-    return (
-        os.path.abspath(Path(__file__).parent / path)
-        .replace(os.path.abspath(os.getcwd()), "")
-        .lstrip("/")
-    )
-
-
-for name in dir(module):
-    value = getattr(module, name)
-    if name.startswith("__"):
-        continue
-
-    if inspect.isclass(value) and (
-        issubclass(value, BlockContext) or issubclass(value, Component)
-    ):
-        file_location = Path(inspect.getfile(value)).parent
-
-        found = [
-            x
-            for x in artifacts
-            if get_relative_path(Path("..") / x)
-            == get_relative_path(file_location / value.TEMPLATE_DIR)
-        ]
-        if len(found) == 0:
-            artifacts.append(
-                os.path.abspath(file_location / value.TEMPLATE_DIR)
-                .replace(os.path.abspath(Path("..")), "")
-                .lstrip("/")
-            )
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Description of your program")
+    parser.add_argument("-m", "--mode", help="Build mode or dev mode")
+    args = parser.parse_args()
+
+    with open("../pyproject.toml") as f:
+        pyproject_source = f.read()
+
+    pyproject_toml = parse(pyproject_source)
+    module_name = pyproject_toml["project"]["name"]
+
+    try:
+        module = importlib.import_module(module_name)
+    except:
+        # The importer will know not to attempt to load
+        # this component if this string is returned
```

Comment:
I don't think it needs to be installed. I tested this with my custom component explicitly uninstalled and not present in `pip list`. Since we are in `/backend` when this script is run the module name will act as a relative import right? Like `import package` will import whatever is in `backend/package`?

Reply:
This was my situation:

```
- component_dir
  - comp1
    - backend
    - frontend
    - pyproject.toml
    - demo
      - app.py
  - comp2
    - backend
    - frontend
    - pyproject.toml
  - comp3
    - backend
    - frontend
    - pyproject.toml
 ```

Reply:
Ah right. I'd actually like this to work without needing the packages installed. Will take a look at this case.

Reply:
Especially the build, I think it is reasonable to expect the packages to be installed in dev mode.